packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -576,17 +576,18 @@ export const ComboBoxMixin = (subclass) =>
 
     /** @private */
     _onEnter(e) {
-      // should close on enter when custom values are allowed, input field is cleared, or when an existing
-      // item is focused with keyboard. If auto open is disabled, under the same conditions, commit value.
-      if (
-        (this.opened || this.autoOpenDisabled) &&
-        (this.allowCustomValue || this._inputElementValue === '' || this._focusedIndex > -1)
-      ) {
-        this._closeOrCommit();
+      // do not commit value when custom values are disallowed and input value is not a valid option
+      if (!this.allowCustomValue && this._inputElementValue !== '' && this._focusedIndex < 0) {
+        return;
```

Comment:
Not stopping propagation in this case is arguable, but the previous logic worked in the same way, regardless whether the combo box was opened or not.

Reply:
We decided to change this after all to stop propagation as well. There is one edge-case where pressing enter could submit a native form, which would then use the invalid value that is still in the input. Similarly Flow/Hilla developers could add an enter shortcut listener to submit the Binder, which would correctly use the previous value, but it would confuse end users who would still see the custom invalid value in the input.